Makefile.build

```diff
@@ -362,9 +362,14 @@ else
   DYNDEP=-dyndep opt
 endif
 
+# can easily produce too long command line with many .v files prefixed by build_vo
+# so we use find + xargs to split coqdep invocation if necessary
+# OSX sed doesn't support -null-data so can't use -print0
 $(VFILED): $(BUILD_VFILES) $(BUILD_MLFILES) $(COQDEP) | all-src
 	$(SHOW)'COQDEP    VFILES'
-	$(HIDE)$(COQDEP) -boot $(DYNDEP) -R $(VO_OUT_DIR)theories Coq -Q $(VO_OUT_DIR)user-contrib "" $(PLUGININCLUDES) $(USERCONTRIBINCLUDES) $(BUILD_VFILES) $(TOTARGET)
+	$(HIDE)find theories $(addprefix user-contrib/, $(USERCONTRIBDIRS)) -type f -name '*.v' | \
+	sed 's|^|$(VO_OUT_DIR)|' | \
+	xargs $(COQDEP) -boot $(DYNDEP) -R $(VO_OUT_DIR)theories Coq -Q $(VO_OUT_DIR)user-contrib "" $(PLUGININCLUDES) $(USERCONTRIBINCLUDES) > "$@"
```

Comment:
:vomiting_face:

Reply:
Do we really have to support Windows quirks in the bridge build? Wasn't dune promising singing tomorrows?

Reply:
Our problem is precisely that we haven't a fully dunerized build yet...

Reply:
We will arrive there @ppedrot , native + dune with coqnative works pretty well IMO, however vos seems pretty tricky, you can have a look at the issues, but I am not sure how to fit a vos build in an incremental setup.

Reply:
(Slight digression: my personal take on that is that the whole vos system is completely screwed from the very beginning. We should instead have a mode where **vo files** can be built skipping the proofs and expose binaries to the build system to query the state of said vo files. How we could believe that relying on the filesystem for vok files is beyond my understanding.)

Reply:
Because make dependencies cannot be that dynamic. Once vos are there al vok can be computed in parallel, while in the usual compilation chain you have to follow coqdep's analysis. The same is true for vio2vo. If you name all kinds of files .vo, you can't tell make to keep or ditch a dependency based on what is in a file. Maybe dune can, I don't know.

Reply:
Dune would have a hard time too, tho it can track the flags used in files so a way to re-trigger a rule is to update the rule itself. Tho the best way to try anything with Dune is to actually try to write the rule in the OCaml language, which is pretty interesting as types will tag the provenance of different things, while abstracting many annoying details.

@ppedrot what you propose could be interesting, but my experience tells me this topic is in general complicated; people is considering these days the intersection of compilers and build systems as a new research topic actually.

IMHO the way to go here [as I said in the vos PR] is to actually propose the build rules _first_, once we are convinced these rules do make sense and provide good incremental builds, you can go ahead an implement them.

Reply:
Please digress on zulip not random prs